File: CodingTest/성호/20주차/인사고과.js
function solution(scores) {
  let wanho = scores[0];
  let answer = 1;
  scores.sort((a, b) => {
    return a[0] === b[0] ? a[1] - b[1] : b[0] - a[0];
  });
  let max = scores[0][1];
  for (let i = 0; i < scores.length; i++) {
    const score = scores[i];
    if (score[1] < max) {
      if (score[0] === wanho[0] && score[1] === wanho[1]) {
        return -1;
      }
    } else {
      max = Math.max(max, score[1]);
      if (score[0] + score[1] > wanho[0] + wanho[1]) {
        answer++;
      }
    }
  }
  return answer;
}

File: CodingTest/성유진/21주차/인사고과.java
import java.util.*;

class Solution {
  // 인센티브를 받을 수 있는지 여부
  boolean isPossible(int cur, int[][] scores) {
    for (int i = 0; i < scores.length; i++) {
      if (scores[cur][0] < scores[i][0] && scores[cur][1] < scores[i][1])
        return false;
    }
    return true;
  }

  public int solution(int[][] scores) {
    // 완호가 인센티브 못 받는 경우 미리 처리
    if (!isPossible(0, scores))
      return -1;
    
    int answer = 0;
    int wanhoTotal = scores[0][0] + scores[0][1];
    // 점수 합계 내림차순 정렬
    Arrays.sort(scores, (a, b) -> {
      return Integer.compare(b[0] + b[1], a[0] + a[1]);
    });
    for (int i = 0; i < scores.length; i++) {
      if (isPossible(i, scores)) {
        int curTotal = scores[i][0] + scores[i][1];
        if (wanhoTotal == curTotal)
          break;
        answer++; // 완호보다 높은 등수의 사람 수 세기
      }
    }
    return answer + 1;
  }
}